pqdb-frontend/src/App.js

```diff
@@ -81,49 +93,77 @@ class App extends React.Component {
     xhr.send();
   }
 
+  switchView(view) {
+    const history = this.props.history;
+    history.replace({
+      pathname: pathname(view)
+    });
+    this.setState({ drawerOpen: false });
+  }
 
+  toggleTheme() {
+    this.setState({ themeId: (this.state.themeId === 'light') ? 'dark' : 'light' });
+  }
 
   render() {
     return (
-      <React.Fragment>
-        <ThemeProvider key={this.state.themeId} theme={themes[this.state.themeId]}>
-          <CssBaseline />
-          {this.state.lightBoxIsOpen && (
-            <Lightbox
-              mainSrc={databaseDiagram}
-              onCloseRequest={() => this.setState({ lightBoxIsOpen: false })}
-
-            />
-          )}
-          <Box p={2}>
-            <Grid container direction="column" spacing={2}>
-              <Grid item>
-                <Box display='flex' justifyContent="center">
-                  <img src={logo} width="500em" height="192.5em" alt="Logo" />
-                </Box>
-              </Grid>
-              {
-                (this.state.db != null) ? <Grid item style={{ overflow: 'auto' }}>{views[this.state.selectedView].view(this.state.db)}</Grid> : <Progess />
-              }
-              <Grid item>
+      <ThemeProvider theme={getTheme(this.state.themeId)}>
+        <CssBaseline />
+        {this.state.lightBoxIsOpen && (
+          <Lightbox
```

Comment:
I think if you display the image below the query field it will be too small to read. I chose the lightbox since it had zooming functionality.
An alternative would be to display the image below the query field and to magnify the content at the cursor position.

Reply:
Something like this: https://ethanselzer.github.io/react-image-magnify/#/react-slick

Reply:
I like!

Another idea (but way more work): split the big diagram into five, one for scheme_*, one for flavor_*, etc. This is the natural way of splitting the diagram in parts: Each of them would just have one arrow which leaves the diagram in the direction of the "one level up"  table (except benchmark where it's two)

Reply:
works well for me! It'd be great if we could set a max height to the image so that it is never larger than say 80% of the screen so the query box is always above, but it doesn't seem to be possible (setting the `style` property sets it on the outer `div`, not on the image)

for really small displays, it's probably not very usable, but that's probably not solvable no matter what (and nobody wants to write queries on a smartphone anyways)